flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackend.java

```diff
@@ -181,7 +182,42 @@
 			keyGroupRange,
 			kvStateRegistry,
 			ttlTimeProvider,
-			new UnregisteredMetricsGroup());
+			new UnregisteredMetricsGroup(),
+			null);
```

Comment:
Will use empty collection and add `@Nonnull` annotation, but since it's not an assertion I guess we still need null checkers before using `stateHandles`?

Reply:
With the annotation you can drop the `null` checks, we consider this as null-free code. That is also the same in other parts of Flink. Furthermore the annotations are also compiled into assertions when running in test/debug mode (enabled asserts) and your IDE will complain and mark it as problem if you try to pass null into an annotated method.